sources/packages/web/src/services/AuthService.ts

```diff
@@ -97,31 +98,19 @@ export class AuthService {
         );
         switch (clientType) {
           case ClientIdType.Student: {
-            await store.dispatch(
-              "student/setStudentProfileData",
-              this.keycloak,
-            );
+            const studentStore = useStudentStore(store);
             const hasStudentAccount =
-              await StudentService.shared.checkStudent();
-            await store.dispatch(
-              "student/setHasStudentAccount",
-              hasStudentAccount,
-            );
+              await StudentService.shared.synchronizeFromUserToken();
+            studentStore.setHasStudentAccount(hasStudentAccount);
```

Comment:
await?

Reply:
I will add it but this one is happening synchronously. I am adding it because the dispatch returns a Promise 😉